xDiffCollection/CollectionBin.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  CollectionBin.swift
+//  CollectionTest
+//
+//  Created by Esteban Garro on 2018-06-05.
+//  Copyright © 2018 aldo. All rights reserved.
+//
+
+import Foundation
+
+internal enum BinOperationType {
```

Comment:
Could we split into separate files all those struct/classes?

Reply:
Sure